ndonnx/_info.py

```diff
@@ -0,0 +1,70 @@
+# Copyright (c) QuantCo 2023-2025
+# SPDX-License-Identifier: BSD-3-Clause
+
+from __future__ import annotations
+
+import ndonnx as ndx
+from ndonnx._array import ndonnx_device
+from ndonnx._data_types import canonical_name
+
+
+class ArrayNamespaceInfo:
+    """Namespace metadata for the Array API standard."""
+
+    _all_array_api_types = [
+        ndx.bool,
+        ndx.float32,
+        ndx.float64,
+        ndx.int8,
+        ndx.int16,
+        ndx.int32,
+        ndx.int64,
+        ndx.uint8,
+        ndx.uint16,
+        ndx.uint32,
+        ndx.uint64,
+    ]
+
+    def capabilities(self) -> dict:
+        return {
+            "boolean indexing": True,
+            "data-dependent shapes": True,
+        }
+
+    def default_device(self):
+        return ndonnx_device
+
+    def devices(self) -> list:
+        return [ndonnx_device]
+
+    def dtypes(
+        self, *, device=None, kind: str | tuple[str, ...] | None = None
+    ) -> dict[str, ndx.CoreType]:
+        # We don't care for device since we are writing ONNX graphs.
+        # We would rather not give users the impression that their arrays
+        # are tied to a specific device when serializing an ONNX graph as
+        # such a concept does not exist in the ONNX standard.
```

Comment:
```suggestion
```
This comment would be better placed in the `NdonnxDevice` class, but we can forego it completely. It is somewhat self-explaining, isn't it?

Reply:
> It is somewhat self-explaining, isn't it?

I think it's self-explanatory for us but I meant it as a note for other contributors since it's an unconventional implementation choice.